frontend/src/app/api/SearchOpportunityAPI.ts

```diff
@@ -44,93 +55,68 @@ export default class SearchOpportunityAPI extends BaseApi {
-    // clear the page query param and set the page back to 1.
-    // On initial load (SearchFetcherActionType.InitialLoad) we honor the page the user sent. There is validation guards
-    // in convertSearchParamstoProperTypes keep 1<= page <= max_possible_page
-    const page_offset =
-      searchInputs.actionType === SearchFetcherActionType.Update &&
-      fieldChanged !== "pagination"
-        ? 1
-        : page;
-
-    const orderByFieldLookup = {
-      opportunityNumber: "opportunity_number",
-      opportunityTitle: "opportunity_title",
-      agency: "agency_code",
-      postedDate: "post_date",
-      closeDate: "close_date",
-    };
-
-    let order_by: PaginationOrderBy = "post_date";
-    if (sortby) {
-      for (const [key, value] of Object.entries(orderByFieldLookup)) {
-        if (sortby.startsWith(key)) {
-          order_by = value as PaginationOrderBy;
-          break; // Stop searching after the first match is found
-        }
+// Translate frontend filter param names to expected backend parameter names, and use one_of syntax
+export const buildFilters = (
+  searchInputs: QueryParamData,
+): SearchFilterRequestBody => {
+  return Object.entries(filterNameMap).reduce(
+    (filters, [frontendFilterName, backendFilterName]) => {
+      const filterData =
+        searchInputs[frontendFilterName as FrontendFilterNames];
+      if (filterData && filterData.size) {
+        filters[backendFilterName] = { one_of: Array.from(filterData) };
+      }
+      return filters;
+    },
+    {} as SearchFilterRequestBody,
+  );
+};
+
+export const buildPagination = (
+  searchInputs: QueryParamData,
+): PaginationRequestBody => {
+  const { sortby, page, fieldChanged } = searchInputs;
+
+  // When performing an update (query, filter, sortby change) - we want to
+  // start back at the 1st page (we never want to retain the current page).
+  // In addition to this statement - on the client (handleSubmit in useSearchFormState), we
+  // clear the page query param and set the page back to 1.
+  // On initial load (SearchFetcherActionType.InitialLoad) we honor the page the user sent. There is validation guards
+  // in convertSearchParamstoProperTypes keep 1<= page <= max_possible_page
+  const page_offset =
+    searchInputs.actionType === SearchFetcherActionType.Update &&
+    fieldChanged !== "pagination"
+      ? 1
+      : page;
+
+  let order_by: PaginationOrderBy = "post_date";
+  if (sortby) {
+    for (const [key, value] of Object.entries(orderByFieldLookup)) {
+      if (sortby.startsWith(key)) {
+        order_by = value as PaginationOrderBy;
+        break; // Stop searching after the first match is found
       }
     }
+  }
 
-    // default to descending
-    let sort_direction: PaginationSortDirection = "descending";
-    if (sortby) {
-      sort_direction = sortby?.endsWith("Desc") ? "descending" : "ascending";
-    }
+  const sort_direction =
```

Comment:
the diff is tough here since I removed these functions from the class, but the only real change I made to the pagination function is here, simplifying the sort_direction logic slightly.